linera-core/src/updater.rs

```diff
@@ -274,18 +273,10 @@ where
                     // synchronize them now and retry.
                     self.send_chain_information_for_senders(chain_id).await?;
                 }
-                Err(NodeError::BlobNotFoundOnRead(_)) if !blob_ids.is_empty() => {
+                Err(NodeError::BlobsNotFound(missing_blob_ids)) => {
                     // For `BlobNotFoundOnRead`, we assume that the local node should already be
                     // updated with the needed blobs, so sending the chain information about the
                     // certificates that last used the blobs to the validator node should be enough.
-                    let missing_blob_ids = stream::iter(mem::take(&mut blob_ids))
-                        .filter(|blob_id| {
-                            let node = self.remote_node.node.clone();
-                            let blob_id = *blob_id;
-                            async move { node.blob_last_used_by(blob_id).await.is_err() }
-                        })
-                        .collect::<Vec<_>>()
-                        .await;
```

Comment:
Is `BlobsNotFound` really guaranteed to contain all missing blobs?
If execution fails because some contract tries to read a missing blob, I assume this error contains only _that_ blob? It can't know about any other blobs that would have been encountered if execution had continued. So maybe we should keep this code here?

Reply:
Good point!